src/chainstate/coordinator/mod.rs

```diff
@@ -0,0 +1,307 @@
+use std::collections::VecDeque;
+use std::thread;
+use std::sync::{Arc, Mutex};
+use std::sync::mpsc::{self, Sender};
+use std::process;
+
+use burnchains::BurnchainHeaderHash;
+use chainstate::burn::{BlockHeaderHash, BlockSnapshot};
+use chainstate::stacks::{StacksBlock, TransactionPayload};
+use chainstate::stacks::db::StacksHeaderInfo;
+use chainstate::stacks::events::{StacksTransactionReceipt};
+
+#[derive(Debug, Clone)]
+struct PoxId;
+
+#[derive(Debug, Clone)]
+pub struct SortitionId (
+    BurnchainHeaderHash,
+    PoxId
+);
+
+#[derive(Debug, Clone)]
+pub struct StacksBlockId {
+    burnchain_header_hash: BurnchainHeaderHash,
+    header_hash: BlockHeaderHash,
+    pox_identifier: PoxId,
+}
+
+struct ChainStateDB;
+impl ChainStateDB {
+
+    pub fn stubbed() -> ChainStateDB {
+        ChainStateDB {}
+    }
+
+    #[allow(unused_variables)]
+    pub fn is_block_processed(&self, block_id: &StacksBlockId) -> Result<bool, ()> {
+        unimplemented!()
+    }
+
+    #[allow(unused_variables)]
+    pub fn process_blocks(&mut self, burnchain_blocks_db: &mut BurnchainBlocksDB, pox_identifier: PoxId) -> Result<Vec<(Option<(StacksHeaderInfo, Vec<StacksTransactionReceipt>)>, Option<TransactionPayload>)>, ()> {
+        unimplemented!()
+    }
+}
+
+struct BurnchainBlocksDB;
```

Comment:
Can you define `BurnchainBlocksDB` and `ChainStateDB` as traits?  Then, we can implement them for `StacksChainState`.